server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -0,0 +1,213 @@
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.ACTION_VALUES;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.PARAM_COMPONENT_ID;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.PARAM_COMPONENT_KEY;
+import static org.sonarqube.ws.client.setting.SettingsWsParameters.PARAM_KEYS;
+
+public class ValuesAction implements SettingsWsAction {
+
+  private static final Splitter DOT_SPLITTER = Splitter.on(".").omitEmptyStrings();
+  private static final Splitter COMMA_SPLITTER = Splitter.on(",");
+
+  private final DbClient dbClient;
+  private final SettingsWsComponentParameters settingsWsComponentParameters;
+  private final PropertyDefinitions propertyDefinitions;
+  private final SettingsLoader settingsLoader;
+
+  public ValuesAction(DbClient dbClient, SettingsWsComponentParameters settingsWsComponentParameters, PropertyDefinitions propertyDefinitions, SettingsLoader settingsLoader) {
+    this.dbClient = dbClient;
+    this.settingsWsComponentParameters = settingsWsComponentParameters;
+    this.propertyDefinitions = propertyDefinitions;
+    this.settingsLoader = settingsLoader;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_VALUES)
+      .setDescription("List settings values.<br>" +
+        "If no value have been set for a setting, then the default value is returned.<br>" +
+        "Either '%s' or '%s' can be provided, not both.<br> " +
+        "Requires one of the following permissions: " +
+        "<ul>" +
+        "<li>'Administer System'</li>" +
+        "<li>'Administer' rights on the specified component</li>" +
+        "</ul>", PARAM_COMPONENT_ID, PARAM_COMPONENT_KEY)
+      .setResponseExample(getClass().getResource("values-example.json"))
+      .setSince("6.1")
+      .setHandler(this);
+    addComponentParameters(action);
+    action.createParam(PARAM_KEYS)
+      .setDescription("List of property keys")
+      .setRequired(true)
+      .setExampleValue("sonar.technicalDebt.hoursInDay,sonar.dbcleaner.cleanDirectory");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    writeProtobuf(doHandle(request), request, response);
+  }
+
+  private ValuesWsResponse doHandle(Request request) {
+    DbSession dbSession = dbClient.openSession(true);
+    try {
+      ComponentDto component = settingsWsComponentParameters.getComponent(dbSession, request);
+      settingsWsComponentParameters.checkAdminPermission(component);
+      Set<String> keys = new HashSet<>(request.mandatoryParamAsStrings(PARAM_KEYS));
+
+      ValuesWsResponse.Builder valuesBuilder = ValuesWsResponse.newBuilder();
+      new SettingsBuilder(dbSession, valuesBuilder, keys, component).build();
+      return valuesBuilder.build();
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private class SettingsBuilder {
```

Comment:
- rename to `ValuesResponseBuilder`
- remove constructor parameter `ValuesWsResponse.Builder`
- return ValuesWsResponse when calling `build()`